src/components/SelectedItemBadge/SelectedItemBadge.tsx

```diff
@@ -71,158 +75,123 @@ export interface Props<PriorityItemValue, ChildrenItemValue>
-                            <Text inline size="small">
-                                {item.label}
-                            </Text>
-                        </ListItem>
-                    ))}
-                </Dropdown>
-            )}
-
-            {children ? (
-                <Dropdown<ChildrenItemValue>
-                    button={
-                        <button
-                            {...elem('optionButton', {
-                                isSelected: dropdownStates.main,
-                            })}
-                            aria-label={buttonLabel}
-                            disabled={isDisabled}
-                            type="button"
-                        >
-                            <Text inline size="small" {...elem('valueText')}>
-                                {label}
-                            </Text>
-                            {additionalLabel && (
-                                <Text {...elem('optionText')} inline size="small">
-                                    {additionalLabel}
+export const SelectedItemBadge = React.forwardRef<HTMLDivElement, Props<string>>(
+    (
+        {
+            children,
+            label,
+            additionalLabel,
+            // isMultiSelect = false,
+            isDisabled = false,
+            onDelete,
+            buttonLabel,
+            deleteButtonLabel,
+            priority,
+            ...rest
+        },
+        ref
+    ) => {
+        const hasPriorityList = priority && priority.list.length > 0;
+
+        const renderPriorityIcon = (priorityType?: Priority, disabled: boolean = false) => {
+            if (!priorityType) {
+                return null;
+            }
+
+            const IconComponent = iconMap[priorityType];
+            return IconComponent ? (
+                <IconComponent
+                    disabled={disabled}
+                    viewBox="0 0 24 24"
+                    {...elem('icon', { [priorityType]: true })}
+                />
+            ) : null;
+        };
+
+        const handleOnDelete = (e: React.KeyboardEvent | React.MouseEvent) => {
+            e.stopPropagation();
+            onDelete?.(e);
+        };
+
+        return (
+            <div {...rest} ref={ref} {...block()}>
+                {hasPriorityList && (
+                    <DropdownRoot>
+                        <DropdownTrigger asChild>
+                            <button
+                                aria-label={priority.buttonLabel}
+                                disabled={isDisabled}
+                                type="button"
+                                {...elem('priorityButton')}
+                            >
+                                {renderPriorityIcon(priority.selectedItem?.priority, isDisabled)}
+                            </button>
+                        </DropdownTrigger>
+                        <DropdownContent sideOffset={5}>
```

Comment:
The `DropdownContent` of the `SelectedItemBadge` must always match the width of the badge and align with it. Previously, we used `refElement={badgeRef}` for this. Currently, it appears misaligned and broken, maybe Radix has something to take care of it 🤔
<img width="214" alt="Screenshot 2024-10-15 at 14 39 57" src="https://github.com/user-attachments/assets/1abcd64b-50b9-482c-96e2-16ad56065604">

<img width="246" alt="Screenshot 2024-10-15 at 14 11 32" src="https://github.com/user-attachments/assets/ed3798aa-cd8f-43bb-b29a-2ba6c7e006e6">